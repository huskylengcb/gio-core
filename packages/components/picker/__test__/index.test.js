import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Picker from '../';

const TabOptions = [
  { tabKey: 'events', name: '选择事件', placeholder: '搜索事件',  options: [{ groupId: 'events', groupName: '选择事件', id: 'a', name: 'a' }]},
  { tabKey: 'variables', name: '选择变量', placeholder: '搜索变量',  options: [{ groupId: 'variables', groupName: '选择变量', id: 'a', name: 'a' }]}
]

describe('Picker unit test', () => {
  const Component = (
    <Picker
      value={[]}
      onChange={jest.fn}
      tabOptions={TabOptions}
      mode='tab'
      valueKey={'id'}
      isMultiple={true}
      onSelect={jest.fn}
      handleChange={jest.fn}
      handleTabChange={jest.fn}
      height={450}
      emptyPlaceholder='没有找到相关结果'
      searchableFields={['name']}
      footer={true}
      onOk={jest.fn}
      onCancel={jest.fn}
    />
  )
  describe('Picker snapshot ', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  })
});
