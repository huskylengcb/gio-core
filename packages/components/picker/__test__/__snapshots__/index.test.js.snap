// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker unit test Picker snapshot  renders correctly 1`] = `
<div
  className="gio-picker-wrapper"
>
  <Dropdown
    getPopupContainer={[Function]}
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    onVisibleChange={[Function]}
    overlay={
      <div
        className="gio-picker__overlay"
      >
        <Tabs
          emptyPlaceholder="没有找到相关结果"
          handleChange={[Function]}
          handleTabChange={[Function]}
          height={450}
          isLoading={false}
          isMultiple={true}
          onChange={[Function]}
          onSelect={[Function]}
          required={false}
          searchableFields={
            Array [
              "name",
            ]
          }
          showSearch={true}
          tabOptions={
            Array [
              Object {
                "name": "选择事件",
                "options": Array [
                  Object {
                    "groupId": "events",
                    "groupName": "选择事件",
                    "id": "a",
                    "name": "a",
                  },
                ],
                "placeholder": "搜索事件",
                "tabKey": "events",
              },
              Object {
                "name": "选择变量",
                "options": Array [
                  Object {
                    "groupId": "variables",
                    "groupName": "选择变量",
                    "id": "a",
                    "name": "a",
                  },
                ],
                "placeholder": "搜索变量",
                "tabKey": "variables",
              },
            ]
          }
          value={Array []}
          valueKey="id"
          width={280}
        />
        <div
          className="gio-picker__sub-panel"
        />
        <ForwardRef>
          <GIOButton
            className=""
            disabled={false}
            loading={false}
            long={false}
            onClick={[Function]}
            rounded={false}
            size="default"
            tooltip=""
            type=""
            value={null}
            withoutBg={false}
          >
            确定
          </GIOButton>
          <Gap
            height={0}
            width={10}
          />
          <GIOButton
            className=""
            disabled={false}
            loading={false}
            long={false}
            onClick={[Function]}
            rounded={false}
            size="default"
            tooltip=""
            type="gray"
            value={null}
            withoutBg={false}
          >
            取消
          </GIOButton>
        </ForwardRef>
      </div>
    }
    placement="bottomLeft"
    prefixCls="gio-picker dropdown ant-dropdown"
    trigger={
      Array [
        "click",
      ]
    }
    visible={false}
  >
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Picker
    </span>
  </Dropdown>
</div>
`;
